### PR TITLE
1.20.2 protocol: nbt data can have different root data types

### PR DESCRIPTION
**Pull request by extremeheat**
After https://github.com/PrismarineJS/prismarine-nbt/pull/79/files

### CODE REVIEW

File: data/pc/1.20.2/protocol.json [hunk 1]
```diff
@@ -28,6 +28,18 @@
     "restBuffer": "native",
     "nbt": "native",
     "optionalNbt": "native",
+    "nbtRoot": [
+      "nbt",
+      {
+        "tagType": "anonymousNbt"
```
Comment (rom1504) [inline data/pc/1.20.2/protocol.json:34]: where is tagType used?

Comment (extremeheat) [reply data/pc/1.20.2/protocol.json:34]: `nbt` is a custom type in node-minecraft-protocol, tagType is passed down to the type serializer code for calling prismarine-nbt parsePacketBuffer

**Comment by wgaylord** [comment]
I think you may have missed a few places where nbt is the type in the protocol to change to the nbtRoot type.  For example in EntityMetaData.  Since all uses of NBT in the protocol now use the anonymous root tag.

File: data/pc/1.20.2/protocol.json [hunk 2]
```diff
@@ -117,7 +129,7 @@
                     },
                     {
                       "name": "nbtData",
-                      "type": "optionalNbt"
+                      "type": "optionalNbtRoot"
```
Comment (frej4189) [inline data/pc/1.20.2/protocol.json:132]: Item NBT data is still written with a named root tag.

**Review by frej4189 — CHANGES_REQUESTED** [review]

File: data/pc/1.20.2/protocol.json (hunk 2 above)
Comment (frej4189) [reply data/pc/1.20.2/protocol.json:132]: Seems I was actually incorrect here. Just very weird naming in Vanilla codebase that had me confused (i.e. `readUnnamedTag` actually reads a named tag, whereas `readAnyTag` reads an unnamed tag)

**Pull request merged**